### PR TITLE
RHCLOUD-26616 Implement notifications-connector-splunk

### PR DESCRIPTION
**Pull request by gwenneg**
This PR introduces the `notifications-connector-splunk` module implementation.

### CODE REVIEW

File: connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/eventing/package-info.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+/**
+ * This package contains classes that were imported from the eventing-integrations
+ * repository with no significant changes made to the original source code.
+ * Major changes will come as part of an upcoming refactoring, after the Splunk
+ * connector has been successfully deployed and tested in production.
+ */
```
Comment (gwenneg) [inline connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/eventing/package-info.java:6]: This should be read before reviewing any class from the `com.redhat.cloud.notifications.connector.splunk.eventing` package.

File: connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/SplunkExceptionProcessor.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import static com.redhat.cloud.notifications.connector.splunk.ExchangeProperty.ACCOUNT_ID;
+
+public class SplunkExceptionProcessor extends ExceptionProcessor {
+
+    private static final String DEFAULT_LOG_MSG = "Message sending failed on %s: [orgId=%s, accountId=%s, historyId=%s, targetUrl=%s]";
+    private static final String HTTP_LOG_MSG = "Message sending failed on %s: [orgId=%s, accountId=%s, historyId=%s, targetUrl=%s] " +
+            "with status code [%d] and body [%s]";
+
+    @Override
+    protected void process(Throwable t, Exchange exchange) {
```
Comment (gwenneg) [inline connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/SplunkExceptionProcessor.java:17]: This method replaces [eventing-integrations/ErrorHandlingRoutes.java#L47-L99](https://github.com/RedHatInsights/eventing-integrations/blob/3faa2fb8c58002ce5473166ade7181ece4fb14ed/splunk-quarkus/src/main/java/com/redhat/console/integrations/ErrorHandlingRoutes.java#L47-L99)

File: connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/SplunkTrustAllManager.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
+package com.redhat.cloud.notifications.connector.splunk;
+
+import org.apache.camel.support.jsse.SSLContextParameters;
+import org.apache.camel.support.jsse.TrustManagersParameters;
+
+import javax.net.ssl.X509TrustManager;
+import java.security.cert.X509Certificate;
+
+public class SplunkTrustAllManager implements X509TrustManager {
```
Comment (gwenneg) [inline connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/SplunkTrustAllManager.java:9]: Inspired from [SplunkTrustAllCACerts.java](https://github.com/RedHatInsights/eventing-integrations/blob/3faa2fb8c58002ce5473166ade7181ece4fb14ed/splunk-quarkus/src/main/java/com/redhat/console/integrations/splunk/SplunkTrustAllCACerts.java) and [SplunkIntegration#L146-L153](https://github.com/RedHatInsights/eventing-integrations/blob/3faa2fb8c58002ce5473166ade7181ece4fb14ed/splunk-quarkus/src/main/java/com/redhat/console/integrations/splunk/SplunkIntegration.java#L146-L153).

File: connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/SplunkRouteBuilder.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import static com.redhat.cloud.notifications.connector.splunk.ExchangeProperty.ACCOUNT_ID;
+import static com.redhat.cloud.notifications.connector.splunk.ExchangeProperty.AUTHENTICATION_TOKEN;
+import static com.redhat.cloud.notifications.connector.splunk.ExchangeProperty.TARGET_URL_NO_SCHEME;
+import static com.redhat.cloud.notifications.connector.splunk.ExchangeProperty.TRUST_ALL;
+import static com.redhat.cloud.notifications.connector.splunk.SplunkTrustAllManager.getSslContextParameters;
+import static org.apache.camel.LoggingLevel.INFO;
+import static org.apache.camel.builder.endpoint.dsl.HttpEndpointBuilderFactory.HttpEndpointBuilder;
+
+@ApplicationScoped
+public class SplunkRouteBuilder extends EngineToConnectorRouteBuilder {
```
Comment (gwenneg) [inline connector-splunk/src/main/java/com/redhat/cloud/notifications/connector/splunk/SplunkRouteBuilder.java:23]: This class is a complete rewriting of [SplunkIntegration.java](https://github.com/RedHatInsights/eventing-integrations/blob/3faa2fb8c58002ce5473166ade7181ece4fb14ed/splunk-quarkus/src/main/java/com/redhat/console/integrations/splunk/SplunkIntegration.java)

**Review by g-duval — APPROVED** [review]
LGTM!
Checked with Gwenneg, some small code improvement may be done in another PR

**Pull request merged**